src/main/scala/beam/agentsim/infrastructure/parking/ParkingZoneFileUtils.scala

```diff
@@ -440,10 +478,28 @@ object ParkingZoneFileUtils extends ExponentialLazyLogging {
       val pricingModel = PricingModel(pricingModelString, newCostInDollarsString)
       val timeRestrictions = parseTimeRestrictions(timeRestrictionsString)
       val chargingPoint = ChargingPointType(chargingTypeString)
-      val numStalls = calculateNumStalls(numStallsString.toDouble)
+      val numStalls = calculateNumStalls(numStallsString.toDouble, reservedFor)
       val parkingZoneIdMaybe =
         if (parkingZoneIdString == null || parkingZoneIdString.isEmpty) None
         else Some(ParkingZone.createId(parkingZoneIdString))
+      val linkMaybe =
+        (if (locationXString == null || locationXString.isEmpty || locationYString == null || locationYString.isEmpty)
+           None
+         else
+           Some(new Coord(locationXString.toDouble, locationYString.toDouble))) match {
+          case Some(coord) if beamServices.isDefined =>
+            Some(
+              NetworkUtils.getNearestLink(beamServices.get.beamScenario.network, beamServices.get.geo.wgs2Utm(coord))
+            )
+          case None if beamServices.isDefined && reservedFor.managerType == VehicleManager.TypeEnum.Household =>
```

Comment:
This branch never takes place because Some(..) is never None. Instead of pattern matching if-else can be used here.

Reply:
If you look at it carefully you will see that the match is for
`(if (locationXString == null || locationXString.isEmpty || locationYString == null || locationYString.isEmpty)
           None
         else
           Some(new Coord(locationXString.toDouble, locationYString.toDouble)))`
I will write it clearly to avoid the confusion

Reply:
You are right. I'm suggesting rewriting this part in the following way:

```
       val locationIsDefined = !StringUtils.isBlank(locationXString) && !StringUtils.isBlank(locationYString)
       val linkMaybe = if (locationIsDefined && beamServices.isDefined) {
        val coord = new Coord(locationXString.toDouble, locationYString.toDouble)
        Some(
          NetworkUtils.getNearestLink(beamServices.get.beamScenario.network, beamServices.get.geo.wgs2Utm(coord))
        )
      } else if (beamServices.isDefined && reservedFor.managerType == VehicleManager.TypeEnum.Household)
        getHouseholdLocation(beamServices.get, reservedFor.managerId.toString) map { homeCoord =>
          NetworkUtils.getNearestLink(
            beamServices.get.beamScenario.network,
            homeCoord
          )
        }
      else
        None
```

Reply:
I'm fine with it as is personally, after you note the subtle difference - except I think the if should be broken into a separate variable, which is then matched against. Or maybe use curly brackets, but I prefer the former. The reason is because this is so easy to mistake as was done already. If you break it apart then the match will be more evident to what it is matching. Beyond that, it is a matter of preference for yours versus Dima's suggestion.